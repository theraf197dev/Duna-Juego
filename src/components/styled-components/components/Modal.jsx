import React from 'react';
import {
    ContainerModalStyles,
    ModalTitleStyles,
    ModalContentStyles,
    ImageStyles,
} from '../styles/Modal.styles.jsx';
import Button from './Button.jsx';

const Modal = ({
    buttonText,
    handleClick,
    image,
    title,
}) => {
  return (
    <ContainerModalStyles>
        <ModalTitleStyles>
            <h1>{title}</h1>
        </ModalTitleStyles>
        <ModalContentStyles>
            <ImageStyles src={image} alt='' />
            <Button buttonText={buttonText} handleClick={handleClick} />
        </ModalContentStyles>
    </ContainerModalStyles>
  );
};

Modal.propTypes = {}

export default Modal;