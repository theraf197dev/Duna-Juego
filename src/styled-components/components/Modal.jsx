import React from 'react';
import {
    ContainerModalStyles,
    ModalTitleStyles,
    ModalContentStyles,
    ImageStyles,
} from '../styles/Modal.styles.jsx';
import Button from './Button.jsx';

const Modal = ({
    buttonText,
    handleSubmit,
    image,
    title,
}) => {
  return (
    <ContainerModalStyles>
        <ModalTitleStyles>
            <h1>{title}</h1>
        </ModalTitleStyles>
        <ModalContentStyles>
            <ImageStyles src={image} alt='' />
            <Button buttonText={buttonText} handleSubmit={handleSubmit} />
        </ModalContentStyles>
    </ContainerModalStyles>
  );
};

Modal.propTypes = {}

export default Modal;