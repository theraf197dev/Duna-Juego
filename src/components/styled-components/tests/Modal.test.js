import { render } from "@testing-library/react";
import Modal from '../components/Modal.jsx';

describe('Modal tests', () => {
  let props = {
    handleChange: jest.fn(),
    id: 'testId',
    value: false,
  };

  it('render component', () => {
      const { container } = render(<Modal {...props} />);

      expect(container).toMatchSnapshot();
  });
});
