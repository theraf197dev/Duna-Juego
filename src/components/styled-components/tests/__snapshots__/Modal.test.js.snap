// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal tests render component 1`] = `
<div>
  <div
    class="sc-bcXHqe cKlEbh"
  >
    <div
      class="sc-gswNZR KNhBV"
    >
      <h1 />
    </div>
    <div
      class="sc-dkrFOg jQa-dgm"
    >
      <img
        alt=""
        class="sc-hLBbgP ditexo"
      />
      <div
        class="sc-eDvSVe iaSCtz"
      >
        <button
          class="sc-jSUZER dZZVJU"
        />
      </div>
    </div>
  </div>
</div>
`;
